cheeseshop/swift.py

```diff
@@ -155,5 +159,17 @@ def __init__(self, keystone_session, region_id, interface='public',
     async def create_tempurl(self, name, container=None):
         assert container is not None or self.container is not None
         container = container or self.container
-
-        # TODO:greghaynes Implement this
+        method = 'PUT'
+        proto, barf, host, version, auth_account = self.endpoint.url.split('/')
+        location = "/".join([version, auth_account, container, name])
+        duration_in_seconds = 60 * 60 * 2
+        expires = int(time() + duration_in_seconds)
+        hmac_body = '{0}\n{1}\n/{2}'.format(method, expires, location)
+        sig = hmac.new(bytes(self.temp_url_key, 'ascii'),
+                       bytes(hmac_body, 'ascii'),
```

Comment:
IMO I would make hmac_body be bytes all the way down. i.e. you can do hmac_body = b'\n'.join([method, expires, location]) on oline 167 if you convert mehod to bytes (method = b'PUT'), etc

At a minimum though, dont explicitly encode to ascii, you should do temp_url_key.encode().